### PR TITLE
feat(api-v3): renew access token with client ID FS-1168

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1168" title="FS-1168" target="_blank">Sub-taskFS-1168</a>  [iOS] Update `POST /access` to API V3
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

In this PR we renew the access token right after the self user client has been registered with backend, if the api version is >= v3. 
Note that we also want to do this for already active sessions when a change to api v3 (or higher) is detected. For this we need the implementation of the api migration manager, which will be done in another PR (https://github.com/wireapp/wire-ios-sync-engine/pull/1814). 

### Dependencies (Optional)

https://github.com/wireapp/wire-ios-transport/pull/261

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[FS-1168]: https://wearezeta.atlassian.net/browse/FS-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FS-1168]: https://wearezeta.atlassian.net/browse/FS-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FS-1168]: https://wearezeta.atlassian.net/browse/FS-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SQPIT-764]: https://wearezeta.atlassian.net/browse/SQPIT-764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Tests/Source/UserSession/ZMUserSessionTests+AccessToken.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if shouldRenew {
+            XCTAssertEqual(transportSession.renewAccessTokenCalls.count, 1)
+            XCTAssertEqual(transportSession.renewAccessTokenCalls.first, "1234abcd")
+        } else {
+            XCTAssertEqual(transportSession.renewAccessTokenCalls.count, 0)
+        }
+
+    }
+    
+}
```
Comment (KaterinaWire) [inline Tests/Source/UserSession/ZMUserSessionTests+AccessToken.swift:60]: Swift lint issue

File: Tests/Source/UserSession/ZMUserSessionTests+AccessToken.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            )
+        }
+    }
+
+    func test_accessTokenRenewalAfterClientRegistration(
+        apiVersion: APIVersion,
+        shouldRenew: Bool
+    ) {
+        // given
+        BackendInfo.apiVersion = apiVersion
```
Comment (netbe) [inline Tests/Source/UserSession/ZMUserSessionTests+AccessToken.swift:42]: is this global? need to be reset in teardown?

**Review by netbe — COMMENTED** [review]
LGTM;) just a question about tests setup

**Review by KaterinaWire — APPROVED** [review]

File: Tests/Source/UserSession/ZMUserSessionTests+AccessToken.swift (hunk 2 above)
Comment (David-Henner) [reply Tests/Source/UserSession/ZMUserSessionTests+AccessToken.swift:47]: I applied a change to reset in the test teardown

**Pull request merged**